daemon/start.go

```diff
@@ -179,7 +182,14 @@ func (daemon *Daemon) containerStart(container *container.Container, checkpoint
 
 	ctx := context.TODO()
 
-	err = daemon.containerd.Create(ctx, container.ID, spec, createOptions)
+	n, err := reference.ParseNormalizedNamed(container.Config.Image)
+	if err != nil {
+		return err
+	}
+	newContainerOpts := []containerd.NewContainerOpts{
+		withImageName(n.String()),
+	}
+	err = daemon.containerd.Create(ctx, container.ID, spec, createOptions, newContainerOpts...)
```

Comment:
Looks like we don't need the `newContainerOpts` (yet?)

```suggestion
	err = daemon.containerd.Create(ctx, container.ID, spec, createOptions, withImageName(n.String()))
```

Reply:
Do you anticipate more options here? Otherwise this is worth considering to change ^^